howto/application/extend.md

```diff
@@ -1,38 +1,45 @@
 
-If you want to extend one application only, use application hooks instead. They are easier to create and do not require creating an addon.
+For both options, you must create one or more hooks first. The options differ in how you add these hooks to your application.
 
-## Extend an application with addons
+<a name="create-hook"></a>
+## Create a hook
 
-See [Create an addon](https://discourse.ubuntu.com/t/creating-an-addon/25284) for detailed instructions.
+A hook is a script file that runs a series of commands at a specific time in the application life cycle. See [Hooks](https://discourse.ubuntu.com/t/hooks/28555) for detailed information.
 
-## Extend an application with hooks
+The general steps for creating a hook are as follows:
 
-The following example shows how to use application hooks to change the Android system locale.
+1. Create or download any files or applications that the hook needs. For example, this could be configuration files or Android applications that you want to run.
+1. In a `hooks/` sub-directory, create a script file with the commands that you want to run. As file name, use the name of the hook.
 
-1. In a new `app` directory, create a `manifest.yaml` file with the following content:
+   [note type="information" status="Tip"]
+   Supported hooks are `pre-start`, `post-start` and `post-stop`. See [Hooks](https://discourse.ubuntu.com/t/hooks/28555).
+   [/note]
+1. Make the hook file executable.
 
-   ```
-   name: custom-locale
-   instance-type: a2.3
-   ```
+You must then add the hook file to your application, either [as an application hook](#application-hooks) or [through an addon](#addon).
 
-1. Build the [CustomLocale](https://android.googlesource.com/platform/development/+/master/apps/CustomLocale) application from the AOSP project and move the compiled APK to the `app` folder.
+The following sections give some examples for hooks.
 
-1. Create the `hooks` directory next to the application manifest file:
+### Create a hook for changing the Android system locale
 
-        mkdir -p hooks
+Complete the following steps to create a hook that changes the Android system locale to Korean:
 
-1. Create a `post-start` script file in the `hooks` directory with the following content:
+1. Build the [CustomLocale](https://android.googlesource.com/platform/development/+/master/apps/CustomLocale) application from the AOSP project and move the compiled APK to a folder of your choice.
+1. In your folder, create a `hooks` sub-directory:
+
+        mkdir -p hooks && cd hooks
+1. In the `hooks` directory, create a `post-start` script file with the following content:
 
    ```
    #!/bin/sh -ex
    if  [ "$CONTAINER_TYPE" = "regular" ]; then
      exit 0
    fi
-   cp "$APP_DIR"/*.apk $ANDROID_ROOTFS/data
+   cp "<hook_dir>"/*.apk $ANDROID_ROOTFS/data
```

Comment:
It's fine to put the apk underneath the hook dir. But if people copy and paste the code snippets, put it into the `post-start` script, it won't work since that requires people to manually change the <hook_dir> to the dir path of their choice.
I'd suggest putting the apk to the top-level of the addon folder and using APP_DIR in the `post-start` script for the simplicity and clarity.

Reply:
I see why we called it `<hook_dir>` now, please see my comment below